src/sql/ColumnType.h

```diff
@@ -10,19 +10,31 @@ namespace hsql {
     LONG,
     FLOAT,
     DOUBLE,
+    DECIMAL,
+    REAL,
     CHAR,
     VARCHAR,
+    VARCHAR_VARYING,
     TEXT,
     DATETIME,
-    DATE
+    DATE,
+    TIME,
   };
 
+  struct DecimalSpecification {
+    DecimalSpecification() = default;
+    DecimalSpecification(int64_t precision, int64_t scale);
+    int64_t precision;
+    int64_t scale;
+};
+
   // Represents the type of a column, e.g., FLOAT or VARCHAR(10)
   struct ColumnType {
     ColumnType() = default;
-    ColumnType(DataType data_type, int64_t length = 0);
+    ColumnType(DataType data_type, int64_t length = 0, DecimalSpecification decimal_specification = DecimalSpecification{});
     DataType data_type;
     int64_t length;  // Used for, e.g., VARCHAR(10)
+    DecimalSpecification decimal_specification; // used for, e.g.,  DECIMAL(6, 4)
```

Comment:
`used for, e.g.,  DECIMAL(6, 4)`, I'd remote the `e.g.` or do other cases exist in which decimal_specification is used for a ColumnType?